pkg/resource/cluster.go

```diff
@@ -105,6 +144,90 @@ func (cluster Cluster) StatefulSetName() string {
 	return cluster.Name()
 }
 
+func (cluster Cluster) JobName() string {
+	return cluster.Name()
+}
+func (cluster Cluster) IsSupportedImage() bool {
+	image := cluster.GetCockroachDBImageName()
+	return !strings.EqualFold(image, NotSupportedVersion)
+}
+func (cluster Cluster) LookupSupportedVersion(version string) (string, bool) {
+	if version == "" {
+		return "", false
+	}
+	supportedVersions := getSupportedCrdbVersions()
+	for _, v := range supportedVersions {
+		if strings.EqualFold(version, v) {
+			return v, true
+		}
+	}
+	return "", false
+}
+
+func (cluster Cluster) GetAnnotationVersion() string {
+	return cluster.getAnnotation(CrdbVersion)
+}
+
+func (cluster Cluster) GetAnnotationContainerImage() string {
+	return cluster.getAnnotation(CrdbContainerImage)
+}
+
+func (cluster Cluster) GetAnnotationHistory() string {
+	return cluster.getAnnotation(CrdbHistory)
+}
+
+func (cluster Cluster) getAnnotation(key string) string {
+	if val, ok := cluster.cr.Annotations[key]; !ok {
+		return ""
+	} else {
+		return val
+	}
+}
+func (cluster Cluster) SetAnnotationVersion(version string) {
+	cluster.cr.Annotations[CrdbVersion] = version
+}
+func (cluster Cluster) SetAnnotationContainerImage(containerimage string) {
+	cluster.cr.Annotations[CrdbContainerImage] = containerimage
+}
+
+func (cluster Cluster) GetCockroachDBImageName() string {
+	supportedImages := getSupportedCrdbImages()
+	if cluster.Spec().CockroachDBVersion != "" {
+		if version, ok := cluster.LookupSupportedVersion(cluster.Spec().CockroachDBVersion); ok {
+			if image, ok := supportedImages[version]; ok {
+				return image
+			}
+		}
+		return NotSupportedVersion
+	}
+
+	//we validate the version after the job runs with exec
+	return cluster.Spec().Image.Name
+}
+
+// getSupportedCrdbImages will dynamic build an slice using the env var added in the operator.yaml
+// We will add all the env var that start with RELATED_IMAGE
+func getSupportedCrdbImages() map[string]string {
```

Comment:
nit: Declare these either at before or after all the cluster methods or in their own file.

Reply:
ok... but this should be also moved in another file and add some tests for this part..